cylc/flow/rundb.py

```diff
@@ -1085,3 +1085,27 @@ def select_jobs_for_datastore(
     def vacuum(self):
         """Vacuum to the database."""
         return self.connect().execute("VACUUM")
+
+    def message_in_db(self, submit, message, timestamp, itask):
+        """Has this message been logged in the DB (task_events table)?
+
+        SQL Query in plain English:
+            Say whether the inner query returns any results:
+
+            If the event is a standard task message (e.g. Submit) then
+            there will be no message, so we chect the event column too.
+        """
+        task_name = itask.tokens['task']
+        cycle = itask.tokens['cycle']
+        stmt = r"""
+            SELECT EXISTS (
+                SELECT * FROM task_events WHERE
+                    name == ?
+                    AND cycle == ?
+                    AND (message == ? OR event == ?)
+                    AND submit_num == ?
+                LIMIT 1
+            )
+        """
+        stmt_args = [task_name, cycle, message, message, submit, timestamp]
+        return bool(self.connect().execute(stmt, stmt_args[:-1]).fetchone()[0])
```

Comment:
Wasn't my suggestion that was the problem was it? Only the missing `OR event == ?`?
```suggestion
                SELECT 1 FROM task_events WHERE
                    name == ?
                    AND cycle == ?
                    AND (message == ? OR event == ?)
                    AND submit_num == ?
            )
        """
        stmt_args = [task_name, cycle, message, message, submit]
        return bool(self.connect().execute(stmt, stmt_args).fetchone()[0])
```

Reply:
Working as expected for me